Userland/Libraries/LibWeb/HTML/WorkerDebugConsoleClient.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022, Ben Abraham <ben.d.abraham@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/RefCounted.h>
+#include <AK/Weakable.h>
+#include <LibJS/Console.h>
+
+namespace Web::HTML {
+
+class WorkerDebugConsoleClient final
+    : public JS::ConsoleClient
+    , public RefCounted<WorkerDebugConsoleClient>
+    , public Weakable<WorkerDebugConsoleClient> {
+public:
+    WorkerDebugConsoleClient(JS::Console& console)
+        : ConsoleClient(console)
+    {
+    }
+
+    virtual void clear() override
+    {
+        dbgln("\033[3J\033[H\033[2J");
+        m_group_stack_depth = 0;
+        fflush(stdout);
+    }
+
+    virtual void end_group() override
+    {
+        if (m_group_stack_depth > 0)
+            m_group_stack_depth--;
+    }
+
+    // 2.3. Printer(logLevel, args[, options]), https://console.spec.whatwg.org/#printer
+    virtual JS::ThrowCompletionOr<JS::Value> printer(JS::Console::LogLevel log_level, PrinterArguments arguments) override
```

Comment:
Let's not put these in the header - make a spearate cpp file instead. See:

- https://github.com/SerenityOS/serenity/blob/master/Userland/Services/WebContent/WebContentConsoleClient.h
- https://github.com/SerenityOS/serenity/blob/master/Userland/Services/WebContent/WebContentConsoleClient.cpp

Reply:
Done